pkg/module_manager/models/modules/global.go

```diff
@@ -194,6 +195,10 @@ func (gm *GlobalModule) executeHook(h *hooks.GlobalHook, bindingType sh_op_types
 		return fmt.Errorf("global hook '%s' failed: %s", h.GetName(), err)
 	}
 
+	if hookResult == nil {
+		return nil
+	}
+
```

Comment:
A rather strange logic is used here,
1. We calculate the result `hookResult, err := h.Execute(h.GetConfigVersion(), bc, "global", prefixedConfigValues, prefixedValues, logLabels)`
2. Next, we check that the result is not null and there is a Usage to display the result.
3. Only after that we check for the error, and if there is one, return it.
4. And after that, we are already trying to fully access the hookResult fields, although theoretically it is possible that we get nil. As a result, instead of a mistake, we will get panic.
5. To avoid panic, we do a check on nil before accessing the hookResult fields. Since we've already checked for errors and they're no longer there, since we've got a hookResult with nil, we'll just exit the function with the result nil.

In a good way, after the calculation, you need to immediately check the error, if there is one, return it. Then check for nil, and then decide what to do, return a new error or return nil. And then process the result.

Reply:
yeah, the logic is strange, or even tightly coupled with what h.Execute() returns, but if there is no possibility of getting nil without errors, does it make sense to check? :)

Reply:
and if we check and get nil for the result, but w/o an error, it looks to me like an error condition :thinking: . I think we'd better return a custom error like 'something went wrong' instead of concealing the fact that we'd got nothing.

Reply:
It makes sense to think about it

Reply:
if we got an nil hookresult, we must get non nil error.
if we not getting non nil error and hookresult is nil - it's a bug on Execute() function.

do not need to double check for nil

Reply:
And I think that double checking won’t hurt here. There may be bugs in the Execute function, and if they appear, there will be panic. Why do we need unnecessary panic at work? Easier to check.

Reply:
1) Unnecessary panic notify us about this bag. With this realisation we skip critical bug.
2) This code change not often to make this check
3) golang has a error handling style. If i see correctly, we not to check anytime and everywhere for nil result from functions in our codebase. Can you link best practice about double checking?

Reply:
If you want to guarantee that we'll never get a nil from Execute() function - you can cover every Execute function with tests in next pr.

Reply:
As you wish :)